app/code/Magento/CustomerGraphQl/Model/Context/AddUserInfoToContext.php

```diff
@@ -48,6 +58,17 @@ public function execute(ContextParametersInterface $contextParameters): ContextP
         $contextParameters->setUserType($currentUserType);
 
         $contextParameters->addExtensionAttribute('is_customer', $this->isCustomer($currentUserId, $currentUserType));
+
+        if ($this->isCustomer($currentUserId, $currentUserType)) {
+            try {
+                $customerGroupId = $this->customerRepository->getById($currentUserId)->getGroupId();
```

Comment:
is this request cached when the previous Context sets the user id? 
it is weird how we do do this check for group but not for the currentUserId

Reply:
I would say no (although not entirely sure). As far as I can tell, if FPC does not return a result (Magento\PageCache\Model\App\FrontController\BuiltinPlugin::aroundDispatch, line 73) processing of the authorization token will always flow in the same way. Otherwise, it would not run at all.

Therefore, line 62 here if ($this->isCustomer...) is probably trying to avoid unnecessary calls to customerRepository->getById(). Reasoning is there may be many requests that are not bearing any tokens, or are bearing invalid ones, and for them we do not want to allocate resources to load entity that we know for sure will not be loaded.

Reply:
I still think this is problematic, and it will probably trigger a significant degradation in query time

Reply:
@cpartica the check for the customer group is now removed.